### PR TITLE
DEVEXP-395: Update Verification SDK from latest specification

### PR DESCRIPTION
**Pull request by asein-sinch**
- Feat: differentiate the ReportResponse from the StatusResponse: remove price, countryId and verificationTimestamp from report responses (and callResult when applicable)
- Feat: add smsOptions and calloutOptions in the start verification request and update the corresponding helpers
- Feat-SDK: propose to set the `expiry` time as a Date
- Tech: refactor duplicated enums

### CODE REVIEW

File: packages/verification/src/models/v1/start-flashcall-verification-response/start-flashcall-verification-response.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@ export interface StartFlashCallVerificationResponse {
   /** Verification identifier used to query for status. */
   id?: string;
   /** The value of the method used for the Verification. For Flashcall Verifications, this will always be `flashCall`. */
-  method?: 'flashCall';
+  method?: 'flashcall';
```
Comment (JPPortier) [inline packages/verification/src/models/v1/start-flashcall-verification-response/start-flashcall-verification-response.ts:8]: 💡 Did you check against server, just to ensure there is not another mix here between `flashCall` and `flashcall` we had in the past

File: packages/verification/src/models/v1/sms-verification-report-response/sms-verification-report-response.ts [hunk 2]
```diff
@@ -1,6 +1,5 @@
-import { VerificationStatusEnum } from '../enums';
+import { ReasonEnum, SourceEnum, VerificationStatusEnum } from '../enums';
 import { Identity } from '../identity';
-import { SMSVerificationReportResponsePrice } from '../verification-report-response-price';
 
 export interface SMSVerificationReportResponse {
```
Comment (JPPortier) [inline packages/verification/src/models/v1/sms-verification-report-response/sms-verification-report-response.ts:4]: Is interface is related to https://developers.sinch.com/docs/verification/api-reference/verification/tag/Verifications/#tag/Verifications/operation/ReportVerificationByIdentity!c=200&path=0/id&t=response ?
Online doc is listing only `id`, `method` and `status` field as response content.

File: packages/verification/src/models/v1/flashcall-verification-status-response/flashcall-verification-status-response.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import { CallResult, ReasonEnum, SourceEnum, VerificationStatusEnum } from '../enums';
+import { Identity } from '../identity';
+import { VerificationPriceCall } from '../verification-price-call';
+
+export interface FlashCallVerificationStatusResponse {
+
+  /** The unique ID of the verification request. */
+  id?: string;
+  /** The method of the verification request. This will always be `flashCall`. */
+  method?: 'flashcall';
```
Comment (JPPortier) [inline packages/verification/src/models/v1/flashcall-verification-status-response/flashcall-verification-status-response.ts:10]: Same here:  doc say `flashCall`
if doc is wrong we should notify them

File: packages/verification/src/models/v1/flashcall-verification-status-response/flashcall-verification-status-response.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  /** The status of the verification request. */
+  status?: VerificationStatusEnum;
+  /** Displays the reason why a verification has `FAILED`, was `DENIED`, or was `ABORTED`. */
+  reason?: ReasonEnum;
+  /** Shows whether the call is complete or not. */
+  callComplete?: boolean;
```
Comment (JPPortier) [inline packages/verification/src/models/v1/flashcall-verification-status-response/flashcall-verification-status-response.ts:16]: Not present onto online doc

File: packages/verification/src/models/v1/flashcall-verification-status-response/flashcall-verification-status-response.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  /** The result of the call. */
+  callResult?: CallResult;
```
Comment (JPPortier) [inline packages/verification/src/models/v1/flashcall-verification-status-response/flashcall-verification-status-response.ts:18]: Not present onto online doc

File: packages/verification/src/models/v1/flashcall-verification-report-response/flashcall-verification-report-response.ts [hunk 6]
```diff
@@ -1,5 +1,4 @@
-import { VerificationReportResponsePrice } from '../verification-report-response-price';
-import { VerificationStatusEnum } from '../enums';
+import { ReasonEnum, SourceEnum, VerificationStatusEnum } from '../enums';
 import { Identity } from '../identity';
 
 export interface FlashCallVerificationReportResponse {
```
Comment (JPPortier) [inline packages/verification/src/models/v1/flashcall-verification-report-response/flashcall-verification-report-response.ts:4]: I'm not finding associated response from online doc synchronized with this declaration 
- start verification: https://developers.sinch.com/docs/verification/api-reference/verification/tag/Verifications/#tag/Verifications/operation/StartVerification!c=200&path=1/id&t=response
- report a verification: https://developers.sinch.com/docs/verification/api-reference/verification/tag/Verifications/#tag/Verifications/operation/ReportVerificationByIdentity!c=200&path=1/id&t=response
- get verificaiton by id: https://developers.sinch.com/docs/verification/api-reference/verification/tag/Verification-status/#tag/Verification-status/operation/VerificationStatusById!c=200&path=1/id&t=response

File: packages/verification/src/models/v1/sms-verification-report-response/sms-verification-report-response.ts (hunk 2 above)
Comment (asein-sinch) [reply packages/verification/src/models/v1/sms-verification-report-response/sms-verification-report-response.ts:4]: And these 3 fields are the one I got from the server. However, in the discussion with the Conversation team, it was mentioned that other properties can be part of this response: https://mailgun.slack.com/archives/C06R5UNSLR0/p1712846563563989

File: packages/verification/src/models/v1/flashcall-verification-status-response/flashcall-verification-status-response.ts (hunk 3 above)
Comment (asein-sinch) [reply packages/verification/src/models/v1/flashcall-verification-status-response/flashcall-verification-status-response.ts:10]: Same behaviour has before: the server sends lowercase:
```json
{
  "id": "018ef61d-9c94-6f13-bd58-ebc87c2883b1",
  "method": "flashcall",
  "status": "FAIL",
  "reason": "Expired",
...
}
```

File: packages/verification/src/models/v1/flashcall-verification-status-response/flashcall-verification-status-response.ts (hunk 4 above)
Comment (asein-sinch) [reply packages/verification/src/models/v1/flashcall-verification-status-response/flashcall-verification-status-response.ts:16]: Then it's a miss from the doc. The server still returns this value:
```json
{
...
  "callComplete": true,
  "callResult": "ANSWERED"
}
```

File: packages/verification/src/models/v1/flashcall-verification-status-response/flashcall-verification-status-response.ts (hunk 5 above)
Comment (asein-sinch) [reply packages/verification/src/models/v1/flashcall-verification-status-response/flashcall-verification-status-response.ts:18]: I've put the 2 fields in the previous response 😁

File: packages/verification/src/models/v1/flashcall-verification-report-response/flashcall-verification-report-response.ts (hunk 6 above)
Comment (asein-sinch) [reply packages/verification/src/models/v1/flashcall-verification-report-response/flashcall-verification-report-response.ts:4]: The schema is defined as a oneOf in the spec: https://gitlab.com/sinch/sinch-projects/enterprise-and-messaging/documentation/developer-experience/newdocs/-/blob/development/docs/verification/api-reference/verification.yaml?ref_type=heads#L895
But I've removed less properties than what is in the current doc, still following the discussion with the backend team

**Review by JPPortier — APPROVED** [review]
Thanks for details

**Pull request merged**